### PR TITLE
[Python] Update to_byte_buff_mapping and from_byte_buff_mapping

### PR DESCRIPTION
**Pull request by jun-he**
Update to_byte_buff_mapping and from_byte_buff_mapping to be consistent and add additional tests.

### CODE REVIEW

**Comment by jun-he** [comment]
@rymurr @TGooch44 Here is the PR to update conversions.
BTW, looks like decimal type is not supported and is commented out in the code. I am wondering if there is any issue with it?

File: python/iceberg/api/types/conversions.py [hunk 1]
```diff
@@ -42,34 +42,34 @@ class Conversions(object):
     to_byte_buff_mapping = {TypeID.BOOLEAN: lambda type_id, value: struct.pack("<h", 1 if value else 0),
                             TypeID.INTEGER: lambda type_id, value: struct.pack("<i", value),
                             TypeID.DATE: lambda type_id, value: struct.pack("<i", value),
-                            TypeID.LONG: lambda type_id, value: struct.pack("<l", value),
-                            TypeID.TIME: lambda type_id, value: struct.pack("<l", value),
-                            TypeID.TIMESTAMP: lambda type_id, value: struct.pack("<l", value),
+                            TypeID.LONG: lambda type_id, value: struct.pack("<q", value),
```
Comment (rymurr) [inline python/iceberg/api/types/conversions.py:45]: why the change here?

**Review by rymurr — COMMENTED** [review]
lgtm, just one comment for clarification.

re decimal: I honestly dont know, my guess is it wasn't tested well enough to be used?

File: python/iceberg/api/types/conversions.py (hunk 1 above)
Comment (jun-he) [reply python/iceberg/api/types/conversions.py:45]: @rymurr  Here, `<q` is for `long long` with 8 bytes and `<i` is for `integer` with 4 bytes.
I think here we need 8 bytes to match Java. Also `from_byte_buff_mapping` uses `<q`.

**Comment by TGooch44** [comment]
> @rymurr @TGooch44 Here is the PR to update conversions.
> BTW, looks like decimal type is not supported and is commented out in the code. I am wondering if there is any issue with it?

It was just never fully tested/implemented.

**Comment by jun-he** [comment]
thanks @TGooch44 @rymurr for the review. I will check the decimal type and put it in a different PR.

File: python/iceberg/api/types/conversions.py (hunk 1 above)
Comment (rymurr) [reply python/iceberg/api/types/conversions.py:45]: ahhh, got it. I thought `long long` meant `double long` not `long` and `l` meant `long`. How bizarre of python...

**Review by rymurr — APPROVED** [review]
